### PR TITLE
Xml vehicles (Menyoo)

### LINKED ISSUE #864: [Request]: INI and XML
### Problem

INI and XML

### Solution

Request the addition of INI and XML support

### Reason

practicability

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Tayipc38**
Initial menyoo xml reader, bound to be updated since I can't be bothered to test it with more files than I did. I've left out what I think was unnecessary.

image
image

Closes #864

### CODE REVIEW

File: src/services/vehicle/xml_vehicles_service.cpp [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+#include "util/vehicle.hpp"
+#include "util/world_model.hpp"
+
+namespace big
+{
+
+	xml_vehicles_service::xml_vehicles_service()
+	{
+		g_xml_vehicles_service = this;
+	}
```
Comment (Yimura) [inline src/services/vehicle/xml_vehicles_service.cpp:15]: No deconstructor present to set the `g_xml_vehicles_service` ptr to `nullptr`.

**Comment by iDominiqe** [comment]
How does it all work? And how do you add yours?

**Comment by iDominiqe** [comment]
works great, thank you very much, waiting for the update! if you can also add support for maps 😍

**Comment by iDominiqe** [comment]
if someone uses a modified xml transport, then errors pop up, they say the user is trying to crash you. Basically works great
image
image
image
image

**Comment by Tayipc38** [comment]
There are missing parts regarding xml nodes, but I believe the essentials are covered. Any incompatible xml files I'd like to see personally so I can add or fix its inclusion.

**Review by Yimura — APPROVED** [review]

**Pull request merged**